docs/adapters/adapter_api.md

```diff
@@ -6,9 +6,65 @@ NOTE: This API is currently under active development and is highly likely to cha
 
 __TOC__
 
+## Overview
+
+An Adapter is the interface between the juttle runtime and a backend data source. The backend consists of a set of data points. Each data point contains a number of fields. In most cases, each data point also contains a timestamp.
+
+For reads, the adapter's job is to interpret the options included in the juttle `read` command into a set of matching data points, construct json objects representing those data points, and pass them to the juttle runtime. For writes, the adapter's job is to take the output of programs and "save" that output by writing data points to the backend.
+
+More sophisticated adapters work together with the juttle optimizer to push aggregation operations directly into the backend. For example, to count the number of data points in a given time period you could simply fetch all the data points and have the juttle program perform the counting. However, if supported by the backend it would be more efficient to count the number of data points directly in the backend and return the count instead.
+
+## Writing an Adapter
+
+Here are the main steps involved in writing a new adapter:
+
+1. Create a javascript module that is require()d when a juttle program executes a `read <adapter>` or `write <adapter>`.
+2. In that module, implement an init() function to define the read/write capabilities of the adapter.
```

Comment:
It doesn't have to be init but it does have to be the export of the module, right? I can reword as "From that module, export an initialization function to define..."